### DISCUSSION TITLE
Async plugin

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/tailwindcss#2081] [Help]
Hi, How can I use a plugin that runs async code?

**Comment by adamwathan** [comment]
Hey! Currently this isn't supported and this is actually the first time anyone has asked about it surprisingly! I think we could support it eventually though, what's your use case?

**Reply by ghost** [reply]
So how can deal with promises? 
I need for working with the file system, dynamic module imports, and some dependencies.

**Reply by ghost** [reply]
@adamwathan Can you help me, please? It's really necessary for me.

**Reply by adamwathan** [reply]
Currently not possible unfortunately. Would need to add it as a feature to Tailwind, maybe in an upcoming release.

**Reply by ghost** [reply]
@adamwathan If it does not provide any breaking changes and you agree, I can work on it and create a PR of course with your guidance.

**Reply by adamwathan** [reply]
Yeah I would happily accept a PR that added support for async plugins if it introduced no breaking changes 👍

**Reply by ghost** [reply]
Sorry, actually it was a question 😅. Do you think it introduces any beaking changes? and where should I start?
